[ES|QL] Update doc for supporting TS and ROW as source commands inside subquery in FROM command (#153021)
elastic/elasticsearch: docs/reference/query-languages/esql/esql-subquery.md

Reviewer: ```suggestion
# Combine result sets with {{esql}} `FROM` subqueries [esql-subquery]
```

slight tweak for SEO

---

> navigation_title: "Combine result sets with subqueries in a FROM command"

Reviewer: ```suggestion
navigation_title: "Use subqueries in a FROM command"
```

this needs to be shorter because it's the left hand nav, so something like this is probably good

Author: I'll modify this, and keep `esql-multi.md` and the H1 unchanged.

applies_to:
  serverless: ga
  stack: preview 9.4, ga 9.5+
products:
  - id: elasticsearch
---

> # {{esql}} subquery [esql-subquery]

Reviewer: ```suggestion
```

we need to delete this because page now has two H1s 😄

Author: Good catch!

---

> navigation_title: "Combine result sets with subqueries"

Reviewer: ```suggestion
navigation_title: "Combine result sets with FROM subqueries"
```

Author: Is it a good idea to change it to "Combine result sets with subqueries in a FROM command"?

highest count (`172.21.3.15` with 4 occurrences) is included when combined with
`employees`.

> ### Combine time series data with a TS subquery

Reviewer: Should any of these headings be tagged as 9.5+ or was this just previously undocumented?

``````markdown
### Combine time series data with a TS subquery
```{applies_to}
stack: ga 9.5+
```
```````

Author: These are new in 9.5, I'll add tags for the sections for `TS` and `ROW`. Thanks for catching it!
